plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -218,16 +218,30 @@ def process_main(self, item=None):
         manifest_type_id = models.Manifest._content_type_id.default
         repo_manifest_ids = repository.get_associated_unit_ids(repo_id, manifest_type_id)
 
-        # check if there is manifest with such id within the queried rpeo
-        manifest = models.Manifest.objects.filter(digest=digest, id__in=repo_manifest_ids)
-        if manifest.count() == 0:
-            raise PulpCodedValidationException(error_code=error_codes.DKR1010,
-                                               digest=digest,
-                                               repo_id=repo_id)
+        # check if there is manifest with such id within the queried repo
+        # since we don't know if the provided digest is of an image manifest or manifest list
+        # we first need to try both
+        #TODO
```

Comment:
I don't have any better suggestions. This might be more livable though if you pulled it out into a separate method. It's a distinct part of the workflow, and that could make it easier to explain and document what it does and why, while isolating the rest of the workflow from its awkwardness.

```
def get_manifest_or_manifest_lists(digest, repo_id, ...):
    ...
    return manifests, manifest_type
```